comment_scan/comment_scanner.rb

```diff
@@ -0,0 +1,274 @@
+require './db'
+require_relative 'message_collection'
+
+class CommentScanner
+    attr_reader :seclient, :chatter, :time_to_scan
+
+    def initialize(seclient, chatter, post_all_comments, ignore_users, perspective_key: '', perspective_log: Logger.new('/dev/null'))
+        @seclient = seclient
+        @chatter = chatter
+        @post_all_comments = post_all_comments
+        @ignore_users = ignore_users
+        @perspective_key = perspective_key
+        @perspective_log = perspective_log
+
+        @time_to_scan = 60
+        @latest_comment_date = @seclient.latest_comment_date.to_i+1 unless @seclient.latest_comment_date.nil?
+    end
+
+    def scan_new_comments
+        new_comments = @seclient.comments_after_date(@latest_comment_date)
+        @latest_comment_date = new_comments[0].json["creation_date"].to_i+1 if new_comments.any? && !new_comments[0].nil?
+        scan_se_comments([new_comments])
+    end
+
+    def tick
+        @time_to_scan -= 1
+        if @time_to_scan <= 0
+            @time_to_scan = 60
+            return false
+        end
+        return true
+    end
+    #def scan_comment(comment_id, should_post_matches: true)
+    #    #Does the work of 
+    #    comment = seclient.comment_with_id(comment_id)
+    #    return false if comment == nil #Didn't actually scan
+    #
+    #    report_comment(comment, should_post_matches: should_post_matches)
+    #end
+
+    def scan_comments_from_db(*comment_ids)
+        comment_ids.flatten.each { |comment_id| scan_comment_from_db(comment_id) }
+    end
+
+    def scan_comment_from_db(comment_id)
+        dbcomment = Comment.find_by(comment_id: comment_id)
+        puts "Found #{dbcomment} for #{comment_id}"
+
+        if dbcomment.nil?
+            @chatter.say("**BAD ID:** No comment exists in the database for id: #{comment_id}")
+            return
+        end
+
+        report_db_comments(dbcomment, should_post_matches: false)
+    end
+
+    def scan_comments(*comment_ids)
+        comment_ids.flatten.each do |comment_id|
+            comment = seclient.comment_with_id(comment_id)
+
+            if comment.nil? #Didn't actually scan
+                @chatter.say("**BAD ID:** No comment exists for id: #{comment_id} (it may have been deleted)")
+                return nil
+            end
+
+            scan_se_comment(comment)#, should_post_matches)
+        end
+    end
+
+    def scan_se_comments(comments)
+        comments.flatten.each { |comment| scan_se_comment(comment) }
+    end
+
+    def scan_se_comment(comment)
+        body = comment.body_markdown
+        toxicity = perspective_scan(body).to_f
+
+        if dbcomment = record_comment(comment, perspective_score: toxicity)
```

Comment:
Could one-line it with `<action> if <condition>`

Reply:
I think I'm going to leave this one just because of how long that line would be (also because I'm using `dbcomment` in the <action> that I'm defining in the <condition>, which imo is a little harder to read when <action> comes first)